cmd/cli/policy_check_conflicts.go

```diff
@@ -0,0 +1,130 @@
+package main
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"strings"
+
+	mapset "github.com/deckarep/golang-set"
+	"github.com/pkg/errors"
+	"github.com/spf13/cobra"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	policyv1alpha1 "github.com/openservicemesh/osm/pkg/apis/policy/v1alpha1"
+	policyClientset "github.com/openservicemesh/osm/pkg/gen/client/policy/clientset/versioned"
+
+	"github.com/openservicemesh/osm/pkg/policy"
+)
+
+const policyCheckConflictsDesc = `
+This command checks whether API resources of the same kind conflict.
+`
+
+const policyCheckConflictsExample = `
+# To check of IngressBackend API resources conflict in the 'test' namespace
+osm policy check-conflicts IngressBackend -n test
+`
+
+type policyCheckConflictsCmd struct {
+	stdout       io.Writer
+	resourceKind string
+	policyClient policyClientset.Interface
+	namespaces   []string
+}
+
+func newPolicyCheckConflicts(stdout io.Writer) *cobra.Command {
+	policyCheckConflictsCmd := &policyCheckConflictsCmd{
+		stdout: stdout,
+	}
+
+	cmd := &cobra.Command{
+		Use:   "check-conflicts RESOURCE_KIND",
+		Short: "check if an API resource conflicts with another resource of the same kind",
+		Long:  policyCheckConflictsDesc,
+		Args:  cobra.ExactArgs(1),
+		RunE: func(_ *cobra.Command, args []string) error {
+			policyCheckConflictsCmd.resourceKind = args[0]
+
+			config, err := settings.RESTClientGetter().ToRESTConfig()
+			if err != nil {
+				return errors.Wrap(err, "Error fetching kubeconfig")
+			}
+
+			policyClient, err := policyClientset.NewForConfig(config)
+			if err != nil {
+				return errors.Wrapf(err, "Error initializing %s client", policyv1alpha1.SchemeGroupVersion)
+			}
+			policyCheckConflictsCmd.policyClient = policyClient
+
+			return policyCheckConflictsCmd.run()
+		},
+		Example: policyCheckConflictsExample,
+	}
+
+	f := cmd.Flags()
+	f.StringSliceVarP(&policyCheckConflictsCmd.namespaces, "namespaces", "n", []string{}, "One or more namespaces to limit conflict checks to")
+
+	return cmd
+}
+
+func (cmd *policyCheckConflictsCmd) run() error {
+	var err error
+
+	switch strings.ToLower(cmd.resourceKind) {
+	case "ingressbackend":
+		err = cmd.checkIngressBackendConflict()
+
+	default:
+		return errors.Errorf("Invalid resource kind %s", cmd.resourceKind)
+	}
+
+	return err
+}
+
+func (cmd *policyCheckConflictsCmd) checkIngressBackendConflict() error {
+	givenNsCount := len(cmd.namespaces)
+	if givenNsCount != 1 {
+		return errors.Errorf("Requires single namespace specified by '-n|--namespace' to check for conflicts, got %d", givenNsCount)
+	}
+
+	ns := cmd.namespaces[0]
+
+	ingressBackends, err := cmd.policyClient.PolicyV1alpha1().IngressBackends(ns).List(context.Background(), metav1.ListOptions{})
```

Comment:
I guess ingress policies, just like smi, could be living in any monitored ns too. Wondering if would be worth to walk all monitored ns for ingress policies

Reply:
thought this could be something more than osm-health could do alternatively maybe

Reply:
Nope, IngressBackend needs to live in the backend's namespace, similar to k8s Ingress, Contour's HTTPProxy etc.

Reply:
> thought this could be something more than osm-health could do alternatively maybe

osm-health is experimental at the moment. We should continue building tooling into osm cli till osm-health is available for general use.